File: 1.1/src/cz/mg/vulkan/PFNvkGetRenderAreaGranularity.java
package cz.mg.vulkan;

public class PFNvkGetRenderAreaGranularity extends VkFunctionPointer {
    public PFNvkGetRenderAreaGranularity() {
    }

    protected PFNvkGetRenderAreaGranularity(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkGetRenderAreaGranularity(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkGetRenderAreaGranularity(long value) {
        setValue(value);
    }

    public PFNvkGetRenderAreaGranularity(VkInstance instance) {
        super(instance, new VkString("vkGetRenderAreaGranularity"));
    }

    public void call(VkDevice device, VkRenderPass renderPass, VkExtent2D pGranularity){
        callNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), renderPass != null ? renderPass.getVkAddress() : VkPointer.getNullAddressNative(), pGranularity != null ? pGranularity.getVkAddress() : VkPointer.NULL);
    }

    protected static native void callNative(long vkaddress, long device, long renderPass, long pGranularity);

}

File: 1.1/src/c/cz/mg/vulkan/PFN_vkGetRenderAreaGranularity.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkGetRenderAreaGranularity_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong renderPass, jlong pGranularity){
    (void)env;
    (void)jc;
    PFN_vkGetRenderAreaGranularity f = (PFN_vkGetRenderAreaGranularity)jniLongToFunctionPointer(address);
    f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkRenderPass*)jniLongToPointer(renderPass)),
        ((VkExtent2D*)jniLongToPointer(pGranularity))
    );
}



